src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/SystemCommandLine/Utils.tt

```diff
@@ -534,4 +530,32 @@ public class OptionData
         GenerateBinderClass(binderClassName, binderTypeClassName, options);
         WriteLine(string.Empty);
     }
+
+    void GenerateRegisterMethodForCommand(string runnerPrefix, XElement commandNode, string binderClassName)
+    {
+        IReadOnlyCollection<OptionData> options = GetOptionsFromNode(commandNode);
+        string customHandlerName = commandNode.Attribute(XName.Get("Handler", ""))?.Value;
+        bool hasHustomHandler = !string.IsNullOrEmpty(customHandlerName);
+
+        string cmdLocalName = "cmd";
+        WriteLine($@"private static void RegisterOptionsForCommand{runnerPrefix}(Command {cmdLocalName}, Func<ILogger> getLogger)");
```

Comment:
Are you using some tool to generate this migration codes? Most of them looks some boilerplate code generated by tool.

Reply:
Yes, this is T4 template code to generate System.CommandLine boilerplate code.

This file is as 'Utils' files for the T4 templates.

The templates that actually generate code are:

- https://github.com/NuGet/NuGet.Client/blob/d36dcd5cb091b499f9cfa513440baed61351d472/src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/SystemCommandLine/CommandParsers.tt 
This template generates a class with a method to populate options to System.CommandLine root command

- https://github.com/NuGet/NuGet.Client/blob/d36dcd5cb091b499f9cfa513440baed61351d472/src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/SystemCommandLine/CustomBinders.tt This template generates boilerplate code to maps command arguments to `**Args` objects, in this case `AddSourceArgs`

- https://github.com/NuGet/NuGet.Client/blob/d36dcd5cb091b499f9cfa513440baed61351d472/src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/SystemCommandLine/VerbParsers.tt
This generates the boilerplate code for each expected option/argument in each command.

Reply:
Added [GeneratedCode] attribute to generated classes.

Reply:
Added readme.md file briefly explaining code generation process.